v2/apiserver/internal/core/events.go

```diff
@@ -549,6 +563,27 @@ func (e *eventsService) CancelMany(
 	// efficient way to do this?
 	go func() {
 		for _, event := range events.Items {
+			// Only iterate through a worker's jobs if the worker status is not
+			// Pending; otherwise no jobs would have been created.
+			if event.Worker.Status.Phase != WorkerPhasePending {
+				for job := range event.Worker.Jobs {
+					if err = e.jobsStore.Cancel(
+						context.Background(),
```

Comment:
Can you just add a comment here that it's deliberate that we're not using the `ctx` that was passed in?